vault/identity_store.go

```diff
@@ -180,100 +185,144 @@ func mfaPaths(i *IdentityStore) []*framework.Path {
 			DisplayAttrs: &framework.DisplayAttributes{
 				OperationPrefix: "mfa",
+				OperationSuffix: methodTypeForOpenIDOperationID + "-method",
 			},
-			Fields: map[string]*framework.FieldSchema{
-				"method_name": {
-					Type:        framework.TypeString,
-					Description: `The unique name identifier for this MFA method.`,
-				},
-				"method_id": {
-					Type:        framework.TypeString,
-					Description: `The unique identifier for this MFA method.`,
-				},
-				"max_validation_attempts": {
-					Type:        framework.TypeInt,
-					Description: `Max number of allowed validation attempts.`,
-				},
-				"issuer": {
-					Type:        framework.TypeString,
-					Description: `The name of the key's issuing organization.`,
-				},
-				"period": {
-					Type:        framework.TypeDurationSecond,
-					Default:     30,
-					Description: `The length of time used to generate a counter for the TOTP token calculation.`,
-				},
-				"key_size": {
-					Type:        framework.TypeInt,
-					Default:     20,
-					Description: "Determines the size in bytes of the generated key.",
-				},
-				"qr_size": {
-					Type:        framework.TypeInt,
-					Default:     200,
-					Description: `The pixel size of the generated square QR code.`,
-				},
-				"algorithm": {
-					Type:        framework.TypeString,
-					Default:     "SHA1",
-					Description: `The hashing algorithm used to generate the TOTP token. Options include SHA1, SHA256 and SHA512.`,
-				},
-				"digits": {
-					Type:        framework.TypeInt,
-					Default:     6,
-					Description: `The number of digits in the generated TOTP token. This value can either be 6 or 8.`,
-				},
-				"skew": {
-					Type:        framework.TypeInt,
-					Default:     1,
-					Description: `The number of delay periods that are allowed when validating a TOTP token. This value can either be 0 or 1.`,
-				},
-			},
+			Fields: methodFieldsIncludingMethodID,
 			Operations: map[logical.Operation]framework.OperationHandler{
 				logical.ReadOperation: &framework.PathOperation{
-					Callback: i.handleMFAMethodTOTPRead,
-					DisplayAttrs: &framework.DisplayAttributes{
-						OperationVerb:   "read",
-						OperationSuffix: "totp-method-configuration|totp-method-configuration",
+					Callback: func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+						return i.handleMFAMethodReadCommon(ctx, req, d, methodType)
 					},
 					Summary: "Read the current configuration for the given MFA method",
 				},
 				logical.UpdateOperation: &framework.PathOperation{
-					Callback: i.handleMFAMethodTOTPUpdate,
+					Callback: func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+						return i.handleMFAMethodWriteCommon(ctx, req, d, methodType)
+					},
 					DisplayAttrs: &framework.DisplayAttributes{
-						OperationVerb:   "configure",
-						OperationSuffix: "totp-method|totp-method",
+						OperationVerb: "update",
```

Comment:
Similar to the above, if this is a configuration we are updating, we've been sticking with the pattern of `configure-x`.

Reply:
I could still see it as `create` a method and `configure` an existing method, but since we are referring to the resource as `x-method` instead of `x-method-configuration`,  `update` probably makes more sense 👍